shapely/strtree.py

```diff
@@ -34,317 +34,185 @@ class STRtree:
+        of geometries that can be joined based on overlapping extents or optional
+        predicate are returned.
 
         Any geometry that is None or empty in the input geometries is omitted from
         the output.
 
         Parameters
         ----------
         geometry : Geometry or array_like
-            Input geometries to query the tree.  The envelope of each geometry
-            is automatically calculated for querying the tree.
+            Input geometries to query the tree and filter results using the
+            optional predicate.
         predicate : {None, 'intersects', 'within', 'contains', 'overlaps', 'crosses',\
 'touches', 'covers', 'covered_by', 'contains_properly', 'dwithin'}, optional
             The predicate to use for testing geometries from the tree
-            that are within the input geometry's envelope.
+            that are within the input geometry's extent.
         distance : number or array_like, optional
             Distances around each input geometry within which to query the tree
             for the 'dwithin' predicate.  If array_like, shape must be
             broadcastable to shape of geometry.  Required if predicate='dwithin'.
 
         Returns
         -------
-        ndarray with shape (2, n)
-            The first subarray contains input geometry indexes.
-            The second subarray contains tree geometry indexes.
+        ndarray with shape (n,) if geometry is a scalar
+            Contains tree geometry indices.
+
+        OR
+
+        ndarray with shape (2, n) if geometry is an array_like
+            The first subarray contains input geometry indices.
+            The second subarray contains tree geometry indices.
 
         Examples
         --------
-        >>> import shapely
-        >>> tree = shapely.STRtree(shapely.points(np.arange(10), np.arange(10)))
-        >>> tree.query_bulk([shapely.box(2, 2, 4, 4), shapely.box(5, 5, 6, 6)]).tolist()
-        [[0, 0, 0, 1, 1], [2, 3, 4, 5, 6]]
-        >>> # Query for geometries that contain tree geometries
-        >>> tree.query_bulk([shapely.box(2, 2, 4, 4), shapely.box(5, 5, 6, 6)], predicate='contains').tolist()
-        [[0], [3]]
-        >>> # To get an array of pairs of index of input geometry, index of tree geometry,
-        >>> # transpose the output:
-        >>> tree.query_bulk([shapely.box(2, 2, 4, 4), shapely.box(5, 5, 6, 6)]).T.tolist()
-        [[0, 2], [0, 3], [0, 4], [1, 5], [1, 6]]
-        >>> # Query for tree geometries within 1 unit distance of input geometries
-        >>> tree.query_bulk([shapely.points(0.5, 0.5)], predicate='dwithin', distance=1.0).tolist()  # doctest: +SKIP
+        >>> from shapely import box, Point
+        >>> import numpy as np
+        >>> points = [Point(0, 0), Point(1, 1), Point(2,2), Point(3, 3)]
+        >>> tree = STRtree(points)
+
+        Query the tree using a scalar geometry:
+        >>> indices = tree.query(box(0, 0, 1, 1))
```

Comment:
```suggestion
        Query the tree using a scalar geometry:

        >>> indices = tree.query(box(0, 0, 1, 1))
```

All the example will need to be updated like this: the `>>>` block needs to be surrounded by a whitespace line to ensure sphinx recognizes it as a code block